.travis.yml

```diff
@@ -0,0 +1,52 @@
+if: (branch = develop) OR (branch = master) OR (type = pull_request) OR (tag IS present)
+language: node_js
+node_js:
+  - "stable"
+cache:
+  directories:
+  - node_modules
+script:
+  - yarn test
+  - yarn build
+after_success:
+  # Pull Request - Deploy it to a review environment
+  # Travis doesn't do deploy step with pull requests builds
+  - ./travis/deploy_pull_request.sh
+deploy:
+  - provider: s3
+    bucket: $DEV_BUCKET_NAME
+    access_key_id: $AWS_ACCESS_KEY_ID
+    secret_access_key: $AWS_SECRET_ACCESS_KEY
+    skip_cleanup: true
+    local_dir: dist
+    on:
+      branch: develop
+
+  - provider: s3
+    bucket: $STAGING_BUCKET_NAME
+    access_key_id: $AWS_ACCESS_KEY_ID
+    secret_access_key: $AWS_SECRET_ACCESS_KEY
+    skip_cleanup: true
+    local_dir: dist
+    upload-dir: current
+    on:
+      branch: master
+
+  # Prepare production deployment
+  - provider: s3
+    bucket: $STAGING_BUCKET_NAME
+    access_key_id: $AWS_ACCESS_KEY_ID
+    secret_access_key: $AWS_SECRET_ACCESS_KEY
+    skip_cleanup: true
+    local_dir: dist
+    upload-dir: releases/$TRAVIS_TAG
+    on:
+      tags: true
+
+  - provider: script
+    script: bash travis/prepare_production_deployment.sh
+    on:
+      tags: true
+on:
```

Comment:
What does this on branch master do?

Reply:
So many deployments for such a simple frontend.

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FBs0B9jBFnU0b6%2Fgiphy.gif&token=toystory" data-gitmeme-token="toystory"><img src="https://media0.giphy.com/media/Bs0B9jBFnU0b6/giphy.gif" title="Created by gitme.me with /toystory"/></a>

Reply:
Deploys, depending on the branch, to different places automatically.
Sends the static code to S3, then is exposed in different URLs

Reply:
This seems to be only active on the master branch though, whereas the deployments specify develop and tagged branches as well.

Can a sub-deployment specification run on e.g. develop if one level up the config is only applied to the master branch?

This part particularly is confusing to me:
```
on:
   branch: master
deploy:
  - provider: ...
     on:
        branch: develop
```

Also, I'd argue we don't need three types of deployments just yet.

Reply:
Yes, I think the first `on` is not right. I need to make tests now once I merge.

It's a matter of effort, for me it's the same setup the environments than only one. It follows the company good practices, and it will help in the future, I understand your skepticism, but for us having different environments was very helpful in the past.